src/System.Private.ServiceModel/src/System/ServiceModel/Description/ClientCredentials.cs

```diff
@@ -27,11 +27,29 @@ protected ClientCredentials(ClientCredentials other)
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("other");
             if (other._userName != null)
                 _userName = new UserNamePasswordClientCredential(other._userName);
+            if (other._clientCertificate != null)
+                _clientCertificate = new X509CertificateInitiatorClientCredential(other._clientCertificate);
+            if (other._serviceCertificate != null)
+                _serviceCertificate = new X509CertificateRecipientClientCredential(other._serviceCertificate);
             if (other._httpDigest != null)
                 _httpDigest = new HttpDigestClientCredential(other._httpDigest);
             _isReadOnly = other._isReadOnly;
         }
 
+        public UserNamePasswordClientCredential UserName
+        {
+            get
+            {
+                if (_userName == null)
+                {
+                    _userName = new UserNamePasswordClientCredential();
+                    if (_isReadOnly)
+                        _userName.MakeReadOnly();
+                }
+                return _userName;
+            }
+        }
+
```

Comment:
This is just a move of locations so it's in the right place (relative to desktop, so it's easier to do a compare)